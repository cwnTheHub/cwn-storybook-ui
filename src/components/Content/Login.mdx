import {  Canvas, Meta, Title, Subtitle, Description, Primary, Controls, Stories } from '@storybook/blocks';

import * as LoginStories from './Login.stories';

<Meta of={LoginStories} />
<Title />
	
`` import { Login } from '@nds_core/nds' ``

Here's a description of the login flow you provided:

1. **User enters username:** The login process begins with the user entering their username into the system.

2. **On input blur, system checks if the user exists:** After the user has input their username and the input field loses focus (e.g., they click or tab out of the username field), the system checks if the entered username exists in its database.

3. **If the user exists, system displays password input for the user:** If the system finds a matching username, it then displays a password input field for the user to enter their password.

4. **User enters password and logs in:** The user enters their password, and upon successful authentication, they are logged into their account.

5. **System checks if the user has subscribed for 2FA:** After successful login with the username and password, the system checks if the user has subscribed to Two-Factor Authentication (2FA).

6. **If the user has subscribed for 2FA, system displays Secure code input for the user:** If the user has enabled 2FA, the system prompts the user to enter a secure code generated by their 2FA method to complete the login process.

7. **If the user has not subscribed for 2FA, the system notifies the user to do so and complete the login process:** If the user has not enabled 2FA, the system informs them of the security benefits and prompts them to set up 2FA to enhance their account security before proceeding.

8. **In case the system checks and the user does not exist, the system displays the registration form:** If the system does not find a matching username during the initial check, it switches to a registration mode, displaying a registration form for the user to complete.

9. **User completes their registration:** The user fills out the registration form with the required information to create a new account.

10. **System must then notify the user to activate their profile to proceed to login:** After successful registration, the system prompts the user to activate their newly created profile. This might involve verifying their email address or taking other necessary activation steps.

11. **If the user's account is not enabled, locked, or expired, the system will display an error message upon login:** If, during any step in the login process, the system detects that the user's account is not enabled, locked, or has expired, it will display an appropriate error message, preventing the user from logging in until the issue is resolved.

This flow describes a comprehensive user authentication and authorization process, including account creation, 2FA considerations, and handling various states of user accounts.

All functions take the following format: 

```json
{
    data: {} || null,
    error:{status:number, message:string} || null,
    success:{status:number, message:string} || null,
}
```

**account with 2FA:** (username: moderator, password: password, secureCode: 123456)

**account without 2FA:** (username: admin, password: password)

<Primary />

## Login component

The component accepts the following inputs (props):

<Controls />

## Additional variations

Listed below are additional variations of the component.

<Stories />