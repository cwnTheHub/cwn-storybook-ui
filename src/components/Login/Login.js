import React from "react";
import Box from "../Box/Box";
import Button from "../Button/Button";
import HBox from "../HBox/HBox";
import HeaderTitle from "../HeaderTitle/HeaderTitle";
import Input from "../Input/Input";
import VBox from "../VBox/VBox";
import { Line, LoginContainer, SubText, Text } from "./LoginStyles";

const Login = ({
  required,
  showConnectWithoutPwd,
  isSecureByCode,
  accountFoundResponse,

  loginHeading,
  txtColor,
  loginEmailLabel,
  loginEmailErrorMessage,
  loginForgotEmailLinkCTA,
  loginEmailHandleChange,
  loginPwdLabel,
  loginPwdErrorMessage,
  loginForgotPwdLinkCTA,
  loginPwdHandleChange,
  loginSecureCodeLabel,
  loginSecureCodeErrorMessage,
  loginSecureCodeHandleChange,
  loginBtnCTATxt,
  loginBtnCTATxtColor,
  loginBtnCTABgColor,
  loginBtnCTAHandleClick,
  loginBtnCTADisable,
  loginBtnCTAWidth,
  loginBtnCTAHeight,
  children,
}) => {
  return (
    <LoginContainer>
      <HBox>
        <VBox>
          <HeaderTitle txtColor={txtColor}>{loginHeading}</HeaderTitle>
          <VBox>
            <Input
              type="email"
              required={required}
              label={loginEmailLabel}
              onChange={loginEmailHandleChange}
              checkEmailValid={true}
              errorMessage={loginEmailErrorMessage}
              forgotUsername={loginForgotEmailLinkCTA}
            />
          </VBox>
          <VBox>
            <Input
              type="password"
              required={required}
              label={loginPwdLabel}
              onChange={loginPwdHandleChange}
              errorMessage={loginPwdErrorMessage}
              forgotPwd={loginForgotPwdLinkCTA}
            />
          </VBox>
          {isSecureByCode && accountFoundResponse && (
            <VBox>
              <Input
                type="text"
                required={required}
                label={loginSecureCodeLabel}
                onChange={loginSecureCodeHandleChange}
                errorMessage={loginSecureCodeErrorMessage}
              />
            </VBox>
          )}
          <VBox>
            <Button
              btnBackgroundColor={loginBtnCTABgColor}
              btnTxtColor={loginBtnCTATxtColor}
              onClick={loginBtnCTAHandleClick}
              disabled={loginBtnCTADisable}
              height={loginBtnCTAHeight}
              width={loginBtnCTAWidth}
              buttonTxt={loginBtnCTATxt}
            />
          </VBox>
        </VBox>
        {showConnectWithoutPwd && <Line />}
        {showConnectWithoutPwd && <VBox>{children}</VBox>}
      </HBox>
    </LoginContainer>
  );
};

export default Login;
