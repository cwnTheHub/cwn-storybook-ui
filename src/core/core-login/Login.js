import React, { useEffect, useState } from "react";
import PropTypes from "prop-types";
import Box from "../core-box/Box";
import Button from "../core-button/Button";
import Card from "../core-card/Card";
import ChevronLink from "../core-chevron-link/ChevronLink";
import FlexGrid from "../core-flex-grid/FlexGrid";
import HairlineDivider from "../core-hairline-divider/HairlineDivider";
import Heading from "../core-heading/Heading";
import Input from "../core-input/Input";
import Paragraph from "../core-paragraph/Paragraph";
import Text from "../core-text/Text";
import { Requirements } from "./requirements/Requirements";
import { getCopy, safeRest } from "../../util-helpers";
import copyDictionary from "./LoginText";
import { pwdRequirementsCopy } from "./requirementText";
import Notification from "../core-notification/Notification";
import TextButton from "../core-text-button/TextButton";

const Login = ({
  variantType,
  cardVariant,
  allowUsernameCheck,
  checkUsernameOrEmailExists,
  sendLoginData,
  send2FALoginData,
  sendSignUPData,
  copy,
  fullHeight,
  spacing,
  contentFootnote,
  response,

  ...rest
}) => {
  const [allowAccontCreation, setAllowAccountCreation] = useState(false);
  const [isVisible, setIsVisible] = useState(false);

  const [username, setUsername] = useState("");
  const [userExists, setUserExists] = useState(false);
  const [usernameIsCkecked, setUsernameIsChecked] = useState(false);
  const [error, setError] = useState(undefined);
  const [status, setStatus] = useState(undefined);

  const [password, setPassword] = useState("");
  const [statusPwd, setStatusPwd] = useState(undefined);
  const [errorPwd, setErrorPwd] = useState(undefined);
  const [confirmPassword, setConfirmPassword] = useState("");
  const [statusConfirmPwd, setStatusConfirmPwd] = useState(undefined);
  const [isValid, setValid] = useState(false);

  const [secureCode, setSecureCode] = useState("");
  const [statusSecureCode, setStatusSecureCode] = useState(undefined);
  const [errorSecureCode, setErrorSecureCode] = useState(undefined);

  const [email, setEmail] = useState(undefined);
  const [statusEmail, setStatusEmail] = useState(undefined);
  const [errorEmail, setErrorEmail] = useState(undefined);

  const [hasErrorOnSignUp, setHasErrorOnSignUp] = useState(false);
  const [has2FA, setHas2FA] = useState(false);

  const [userNotExistAndContinueToRegister, setNextStep] = useState(false);
  const [accountCreationHasFailed, setAccountCreationHasFailed] =
    useState(false);
  const [accountCreationHasSucceed, setAccountCreationHasSucceed] =
    useState(false);

  const [loginHasFailed, setLoginHasFailed] = useState(false);

  const [errorMessage, setErrorMessage] = useState(null);
  const [successMessage, setSuccessMessage] = useState(null);

  const content = getCopy(copyDictionary, copy);

  const emailPattern = /^[a-zA-Z0-9._-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,4}$/;

  const { passwordRequirements } = getCopy(pwdRequirementsCopy, copy);

  useEffect(() => {
    const timer = setTimeout(() => {
      setIsVisible(false);
    }, 10000);

    return () => {
      clearTimeout(timer);
    };
  }, [has2FA, setHas2FA]);

  const onDataFilling = (e) => {
    let text = e?.target?.name;
    switch (text) {
      case "username":
        setUsernameIsChecked(false);
        setUsername(e.target.value);
        if (username.length > 3) {
          setError(undefined);
          setStatus(undefined);
        }
        break;
      case "password":
        setPassword(e.target.value);
        if (password?.length > 0 && password.length < 8) {
          setValid(false);
          setErrorPwd(undefined);
        } else if (isValid) {
          setErrorPwd(undefined);
          setStatusPwd(undefined);
        }
        break;
      case "confirmPassword":
        setConfirmPassword(e.target.value);
        break;
      case "email":
        setEmail(e.target.value);
        if (emailPattern.test(email)) {
          setStatusEmail(undefined);
          setErrorEmail(undefined);
        }
        break;
      case "secureCode":
        setSecureCode(e.target.value);
        break;
      default:
    }
  };

  const validate = (event) => {
    const value = event.target.value;
    let text = event?.target?.name;

    switch (text) {
      case "username":
        if (value.length <= 3) {
          setError(content?.emptyField);
          setStatus("error");
          setUserExists(false);
        } else if (allowUsernameCheck) {
          setError(undefined);
          setStatus(undefined);
          setUsernameIsChecked(false);
          checkUsernameOrEmailExists(username);
          const { data, error } = response;
          if (!error) {
            setUserExists(true);
          }
          if (allowAccontCreation) {
            if (!error) {
              setStatus("error");
            } else {
              setStatus("success");
              setUsernameIsChecked(true);
            }
          }
        }
        break;
      case "email":
        if (!emailPattern.test(email)) {
          setStatusEmail("error");
          setErrorEmail(content?.emailErrorTxt);
        } else {
          setStatusEmail(undefined);
          setErrorEmail(undefined);
        }
        break;
      case "password":
        if (!value.length) {
          setErrorPwd(content?.emptyField);
          setStatusPwd("error");
        } else {
          setErrorPwd(undefined);
          setStatusPwd(undefined);
        }
        break;
      case "confirmPassword":
        if (confirmPassword != password || confirmPassword?.length < 8) {
          setStatusConfirmPwd("error");
        } else {
          setStatusConfirmPwd("success");
        }
        break;
      case "secureCode":
        if (secureCode?.length != 6) {
          setStatusSecureCode("error");
          setErrorSecureCode(content?.incorrectLengthSecureCodeTxt);
        } else {
          setStatusSecureCode(undefined);
          setErrorSecureCode(undefined);
        }

        break;
      default:
    }
  };

  const onContinueCTA = (e) => {
    e.preventDefault();
    setSuccessMessage(null);
    setErrorMessage(null);
    setNextStep(true);
  };

  const signUpCTA = (e) => {
    e.preventDefault();
    setSuccessMessage(null);
    setErrorMessage(null);

    if (
      statusConfirmPwd == "success" &&
      isValid &&
      !errorEmail &&
      !userExists
    ) {
      sendSignUPData({ username, email, password });
      const { data, success, error } = response;
      if (success && data) {
        //set success
        setAccountCreationHasSucceed(!accountCreationHasSucceed);
      }
      if (error) {
        //set Error
        setAccountCreationHasFailed(!accountCreationHasFailed);
      }
    } else {
      setHasErrorOnSignUp(!hasErrorOnSignUp);
    }
  };

  const signInCTA = (e) => {
    e.preventDefault();
    setSuccessMessage(null);
    setErrorMessage(null);

    if (password.length < 1) {
      setErrorPwd(content?.emptyField);
      setStatusPwd("error");
    } else if (username?.length < 3) {
      setError(content?.emptyField);
      setStatus("error");
    } else {
      sendLoginData({ username, password });

      const { data, error, success } = response;
      const has2FAActive = data?.has2FARegistered;

      if (has2FAActive) {
        // display secure code
        setHas2FA(true);
        setIsVisible(true);
        setLoginHasFailed(false);
      }
      if (error) {
        //set Error
        setLoginHasFailed(true);
      }
    }
  };
  const submitSecureCodeCTA = (e) => {
    e.preventDefault();
    setSuccessMessage(null);
    setErrorMessage(null);

    send2FALoginData({
      username,
      password,
      secureCode,
    });
    const { error } = response;
    if (error) {
      // display error
      setLoginHasFailed(true);
    } else {
      setLoginHasFailed(false);
    }
  };

  const renderGeneralError = () => {
    return (
      <Notification
        variant="error"
        copy={copy}
        data-testid="notification-error"
      >
        <Text small> {content?.errorOnSecureCodeTxt}</Text>
      </Notification>
    );
  };

  const renderExistingAccount = () => {
    return (
      <Notification
        variant="warning"
        copy={copy}
        data-testid="account-existing-warning"
      >
        <Text small> {content?.onSuccessHeading}</Text>
      </Notification>
    );
  };
  const renderEmailSentToUser = () => {
    return (
      <Notification
        variant="success"
        copy={copy}
        data-testid="email-sent-notification"
      >
        <Text small> {content?.emailSentTxt}</Text>
      </Notification>
    );
  };
  const renderErrorOnSignUp = () => {
    return (
      <Notification
        variant="error"
        copy={copy}
        data-testid="account-creation-error"
      >
        <Text small> {content?.signUpError}</Text>
      </Notification>
    );
  };
  const renderGeneralErrorOnCreation = () => {
    return (
      <Notification
        variant="error"
        copy={copy}
        data-testid="notification-error"
      >
        <Text small> {content?.onAccountCreatedFailed}</Text>
      </Notification>
    );
  };

  const allowAccontCreationCTA = (e) => {
    e.preventDefault();
    setAllowAccountCreation(!allowAccontCreation);
    setError(undefined);
    setStatus(undefined);
    setErrorPwd(undefined);
    setStatusPwd(undefined);
  };

  if (accountCreationHasSucceed) {
    return (
      <FlexGrid>
        <FlexGrid.Row>
          <FlexGrid.Col xs={12} md={8}>
            <Card
              variant={cardVariant}
              {...safeRest(rest)}
              fullHeight={fullHeight}
            >
              <Box between={3}>
                <Notification
                  variant="success"
                  copy={copy}
                  data-testid="account-createion-success"
                >
                  <Text small> {content?.onAccountCreatedSuccess}</Text>
                </Notification>
              </Box>
            </Card>
          </FlexGrid.Col>
        </FlexGrid.Row>
      </FlexGrid>
    );
  }

  if (has2FA) {
    return (
      <FlexGrid>
        <FlexGrid.Row>
          <FlexGrid.Col xs={12} md={8}>
            <Card
              variant={cardVariant}
              {...safeRest(rest)}
              fullHeight={fullHeight}
            >
              <Box between={3}>
                <>
                  {loginHasFailed ? renderGeneralError() : null}
                  {isVisible ? renderEmailSentToUser() : null}
                </>
                <>
                  <Input
                    type={"number"}
                    hintPosition={"below"}
                    label={content?.secureCode}
                    name="secureCode"
                    autocomplete={true}
                    onChange={onDataFilling}
                    onBlur={validate}
                    value={secureCode}
                    feedback={statusSecureCode}
                    error={errorSecureCode}
                    data-testid="secure-code-input"
                  />
                  {secureCode?.length == 6 ? (
                    <Button
                      onClick={submitSecureCodeCTA}
                      variant="primary"
                      data-testid="sign-in-with-code-button"
                    >
                      {content?.submitSecureCodeCTATxt}
                    </Button>
                  ) : null}
                </>
                <Box data-testid="resend-code">
                  <TextButton onClick={signInCTA}>
                    {content?.resendSecureCodeTxt}
                  </TextButton>
                </Box>
                <div data-testid="create-a-ticket">
                  <ChevronLink href="#">{content?.createTicket}</ChevronLink>
                </div>
              </Box>
            </Card>
          </FlexGrid.Col>
        </FlexGrid.Row>
      </FlexGrid>
    );
  }
  if (!allowAccontCreation && variantType == "regular") {
    return (
      <FlexGrid>
        <FlexGrid.Row>
          <FlexGrid.Col xs={12} md={8}>
            <Card
              variant={cardVariant}
              {...safeRest(rest)}
              fullHeight={fullHeight}
            >
              <Box between={3}>
                <>
                  <Heading level="h3">{content?.heading}</Heading>
                  <Text size="medium">{content?.subtext}</Text>
                  <Box inline between={2}>
                    <Paragraph
                      size={"small"}
                      data-testid="set-up-a-new-account"
                    >
                      {content?.doNotHaveAccountTxt}{" "}
                      <TextButton onClick={allowAccontCreationCTA}>
                        {content?.createAccountTxt}
                      </TextButton>
                    </Paragraph>
                  </Box>

                  {loginHasFailed ? renderGeneralError() : null}
                  <HairlineDivider />
                </>
                <>
                  <Input
                    type={"text"}
                    hintPosition={"below"}
                    label={content?.username}
                    name="username"
                    autocomplete={true}
                    onChange={onDataFilling}
                    onBlur={validate}
                    value={username}
                    feedback={status}
                    error={error}
                    data-testid="username"
                  />
                  <Input
                    type={"password"}
                    hintPosition={"below"}
                    label={content?.password}
                    name="password"
                    autocomplete={false}
                    onChange={onDataFilling}
                    onBlur={validate}
                    value={password}
                    feedback={statusPwd}
                    error={errorPwd}
                    data-testid="pwd"
                  />

                  <Button
                    onClick={signInCTA}
                    variant="primary"
                    data-testid="sign-in-button"
                  >
                    {content?.signInCTA}
                  </Button>
                  <Box inline between={4}>
                    <div data-testid="link-pwd-forgot">
                      <ChevronLink href="#">
                        {content?.pwdForgotTxt}
                      </ChevronLink>
                    </div>
                    <span>|</span>
                    <div data-testid="link-username-forgot">
                      <ChevronLink href="#">
                        {content?.usernameForgotTxt}
                      </ChevronLink>
                    </div>
                  </Box>
                </>
                <div data-testid="create-a-ticket">
                  <ChevronLink href="#">{content?.createTicket}</ChevronLink>
                </div>
              </Box>
            </Card>
          </FlexGrid.Col>
        </FlexGrid.Row>
      </FlexGrid>
    );
  }
  if (
    allowAccontCreation &&
    variantType == "regular" &&
    userNotExistAndContinueToRegister
  ) {
    return (
      <FlexGrid>
        <FlexGrid.Row>
          <FlexGrid.Col xs={12} md={8}>
            <Card
              variant={cardVariant}
              {...safeRest(rest)}
              fullHeight={fullHeight}
            >
              <Box between={3}>
                <>
                  {hasErrorOnSignUp ? (
                    <>
                      {renderErrorOnSignUp()} <HairlineDivider />
                    </>
                  ) : null}
                  {accountCreationHasFailed ? (
                    <>
                      {renderGeneralErrorOnCreation()} <HairlineDivider />
                    </>
                  ) : null}
                  <Input
                    type="email"
                    hintPosition={"below"}
                    hint={content?.fieldRequiered}
                    label={content?.email}
                    name="email"
                    onChange={onDataFilling}
                    onBlur={validate}
                    value={email}
                    feedback={statusEmail}
                    autocomplete={true}
                    error={errorEmail}
                    data-testid="email-input"
                  />
                  {usernameIsCkecked && password?.length && !isValid ? (
                    <Requirements
                      value={password}
                      requirements={passwordRequirements}
                      onValidChange={(isValid) => setValid(isValid)}
                    />
                  ) : null}
                  <Input
                    type={"password"}
                    hintPosition={"below"}
                    hint={content?.fieldRequiered}
                    label={content?.password}
                    name="password"
                    onChange={onDataFilling}
                    onBlur={validate}
                    value={password}
                    feedback={statusPwd}
                    error={errorPwd}
                    data-testid="password-input"
                  />
                  <Input
                    type={"password"}
                    hintPosition={"below"}
                    label={content?.confirmPassword}
                    name="confirmPassword"
                    onChange={onDataFilling}
                    onBlur={validate}
                    value={confirmPassword}
                    feedback={statusConfirmPwd}
                    autocomplete="off"
                    data-testid="confirm-pwd-input"
                  />
                </>
                <Button
                  onClick={signUpCTA}
                  variant="primary"
                  data-testid="sign-up-button"
                >
                  {content?.signUpCTA}
                </Button>
                <div data-testid="create-a-ticket">
                  <ChevronLink href="#">{content?.createTicket}</ChevronLink>
                </div>
              </Box>
            </Card>
          </FlexGrid.Col>
        </FlexGrid.Row>
      </FlexGrid>
    );
  }
  if (allowAccontCreation && variantType == "regular") {
    return (
      <FlexGrid>
        <FlexGrid.Row>
          <FlexGrid.Col xs={12} md={8}>
            <Card
              variant={cardVariant}
              {...safeRest(rest)}
              fullHeight={fullHeight}
            >
              <Box between={3}>
                <>
                  <Heading level="h3">{content?.heading}</Heading>
                  <Text size="medium">{content?.subtext}</Text>
                  <Paragraph size={"small"}>{content?.paragraph}</Paragraph>
                  <Box inline between={2}>
                    <Paragraph
                      size={"small"}
                      data-testid="set-up-a-new-account"
                    >
                      {content?.haveAccountTxt}{" "}
                      <TextButton
                        data-testid="create-text-button"
                        onClick={allowAccontCreationCTA}
                      >
                        {content?.connexionTxt}
                      </TextButton>
                    </Paragraph>
                  </Box>
                  {userExists && renderExistingAccount()}
                  <HairlineDivider />
                </>
                <>
                  <Input
                    type={"text"}
                    hintPosition={"below"}
                    label={content?.username}
                    name="username"
                    autocomplete={true}
                    onChange={onDataFilling}
                    onBlur={validate}
                    value={username}
                    feedback={status}
                    error={error}
                    data-testid="username"
                  />

                  {!userExists && status !== "error" && usernameIsCkecked ? (
                    <Button
                      onClick={onContinueCTA}
                      variant="primary"
                      data-testid="create-account-button"
                    >
                      {content?.continueCTA}
                    </Button>
                  ) : null}
                </>
                <div data-testid="create-a-ticket">
                  <ChevronLink href="#">{content?.createTicket}</ChevronLink>
                </div>
              </Box>
            </Card>
          </FlexGrid.Col>
        </FlexGrid.Row>
      </FlexGrid>
    );
  }

  if (variantType == "inHouse") {
    return (
      <FlexGrid>
        <FlexGrid.Row>
          <FlexGrid.Col xs={12} md={8}>
            <Card
              variant={cardVariant}
              {...safeRest(rest)}
              fullHeight={fullHeight}
            >
              <Box between={3}>
                <>{userExists ? renderExistingAccount() : null}</>
                <>
                  <Input
                    type={"text"}
                    hintPosition={"below"}
                    label={content?.employeeId}
                    name="username"
                    autocomplete={true}
                    onChange={onDataFilling}
                    onBlur={validate}
                    value={username}
                    feedback={status}
                    error={error}
                    data-testid="username"
                  />

                  <Input
                    type={"password"}
                    hintPosition={"below"}
                    label={content?.password}
                    name="password"
                    autocomplete={false}
                    onChange={onDataFilling}
                    onBlur={validate}
                    value={password}
                    feedback={statusPwd}
                    error={errorPwd}
                    data-testid="pwd"
                  />

                  <Button
                    onClick={signInCTA}
                    variant="primary"
                    data-testid="sign-in-button"
                  >
                    {content?.signInCTA}
                  </Button>
                </>
                <div data-testid="create-a-ticket">
                  <ChevronLink href="#">{content?.createTicket}</ChevronLink>
                </div>
              </Box>
            </Card>
          </FlexGrid.Col>
        </FlexGrid.Row>
      </FlexGrid>
    );
  }
  return null;
};

Login.propTypes = {
  variantType: PropTypes.oneOf(["regular", "inHouse"]).isRequired,
  cardVariant: PropTypes.oneOf([
    "white",
    "lavender",
    "grey",
    "default",
    "branded",
    "alternative",
    "defaultWithBorder",
    "defaultOnlyBorder",
  ]),
  allowUsernameCheck: PropTypes.bool,
  checkUsernameOrEmailExists: PropTypes.func,
  sendLoginData: PropTypes.func,
  send2FALoginData: PropTypes.func,
  sendSignUPData: PropTypes.func,
  response: PropTypes.shape({
    data: PropTypes.object || null,
    error: PropTypes.shape({
      status: PropTypes.number,
      message: PropTypes.string,
    }),
    success: PropTypes.shape({
      status: PropTypes.number,
      message: PropTypes.string,
    }),
  }),
  contentFootnote: PropTypes.array,
  copy: PropTypes.oneOfType([
    PropTypes.oneOf(["en", "fr"]),
    PropTypes.shape({
      heading: PropTypes.string,
      onSuccessHeading: PropTypes.string,
      subtext: PropTypes.string,
      paragraph: PropTypes.string,
      username: PropTypes.string,
      password: PropTypes.string,
      email: PropTypes.string,
      confirmPassword: PropTypes.string,
      continueCTA: PropTypes.string,
      signUpCTA: PropTypes.string,
      signInCTA: PropTypes.string,
      securityCodeCTA: PropTypes.string,
      createTicket: PropTypes.string,
    }),
  ]).isRequired,

  fullHeight: PropTypes.bool,
  spacing: PropTypes.oneOf(["default", "narrow", "compact", "intermediate"]),
};

Login.defaultProps = {
  variantType: "regular",
  cardVariant: "defaultWithBorder",
  copy: "en",
  fullHeight: false,
};
export default Login;
