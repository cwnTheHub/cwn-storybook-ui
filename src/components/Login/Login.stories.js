import React from "react";
import { Link } from "react-router-dom";
import Login from "./Login";

export default {
  title: "Login compnent",
  component: Login,
};

const Template = (args) => {
  return (
    <Login
      loginHeading={args.loginHeading}
      txtColor={args.txtColor}
      loginEmailLabel={args.loginEmailLabel}
      loginEmailErrorMessage={args.loginEmailErrorMessage}
      loginForgotEmailLinkCTA={args.loginForgotEmailLinkCTA}
      loginEmailHandleChange={args.loginEmailHandleChange}
      loginPwdLabel={args.loginPwdLabel}
      loginPwdErrorMessage={args.loginPwdErrorMessage}
      loginForgotPwdLinkCTA={args.loginForgotPwdLinkCTA}
      loginPwdHandleChange={args.loginPwdHandleChange}
      loginSecureCodeLabel={args.loginSecureCodeLabel}
      loginSecureCodeErrorMessage={args.loginSecureCodeErrorMessage}
      loginSecureCodeHandleChange={args.loginSecureCodeHandleChange}
      loginBtnCTATxt={args.loginBtnCTATxt}
      loginBtnCTATxtColor={args.loginBtnCTATxtColor}
      loginBtnCTABgColor={args.loginBtnCTABgColor}
      loginBtnCTAHandleClick={args.loginBtnCTAHandleClick}
      loginBtnCTADisable={args.loginBtnCTADisable}
      loginBtnCTAWidth={args.loginBtnCTAWidth}
      loginBtnCTAHeight={args.loginBtnCTAHeight}
      required={args.required}
      showConnectWithoutPwd={args.showConnectWithoutPwd}
      isSecureByCode={args.isSecureByCode}
      accountFoundResponse={args.accountFoundResponse}
      loginHeadingRight={args.loginHeadingRight}
      loginSubHeadingRight={args.loginSubHeadingRight}
    />
  );
};

export const LoginWithSide = Template.bind({});
LoginWithSide.args = {
  loginHeading: "Log in to My Nemeton",
  txtColor: "green",
  loginEmailLabel: "Email/Username",
  loginEmailErrorMessage: "Please fill out this field",
  loginForgotEmailLinkCTA: <Link to={"#"}>Forgot your username?</Link>,
  loginEmailHandleChange: (param) => console.log(param),
  loginPwdLabel: "Password",
  loginPwdErrorMessage: "Please fill out this field",
  loginForgotPwdLinkCTA: "Forgot your password?",
  loginPwdHandleChange: (param) => console.log(param),
  loginSecureCodeLabel: "Security code",
  loginSecureCodeErrorMessage: "Please fill out this field",
  loginSecureCodeHandleChange: (param) => console.log(param),
  loginBtnCTATxt: "Log in",
  loginBtnCTATxtColor: "#FFFFFF",
  loginBtnCTABgColor: "green",
  loginBtnCTAHandleClick: (param) => console.log(param),
  loginBtnCTADisable: false,
  loginBtnCTAWidth: "",
  loginBtnCTAHeight: "",

  loginHeadingRight:"Tired of remembering passwords? Log in without one.",
  loginSubHeadingRight:"Get an email with a link that instantly and securely logs you into your account.",
  required: true,
  showConnectWithoutPwd: true,
  isSecureByCode: true,
  accountFoundResponse: true,
};
