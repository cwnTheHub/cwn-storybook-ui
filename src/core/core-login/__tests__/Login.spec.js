import React from "react";
import { shallow, mount, render } from "enzyme";
import Login from "../Login";

describe("<Login/>", () => {
  const doShallow = (props = {}) => shallow(<Login {...props} />);
  const doMount = (props = {}) => mount(<Login {...props} />);

  it("renders", () => {
    const login = render(<Login>Children</Login>);

    expect(login).toMatchSnapshot();
  });
});
