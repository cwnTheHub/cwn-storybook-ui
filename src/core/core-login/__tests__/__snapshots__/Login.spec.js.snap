// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login/> renders 1`] = `
<div
  class="sc-feUZmu gLPHiS"
>
  <div
    class="sc-dAbbOL WHwBZ"
  >
    <div
      class="sc-ikkxIA bJfcxL"
    >
      <div
        class="sc-aXZVg hgmtPE sc-gsFSXq cGRQzy sc-gsFSXq cGRQzy"
      >
        <div
          class="sc-aXZVg cPJwEj"
        >
          <h3
            class="sc-hzhJZQ ejsfGD"
            data-testid="heading"
          >
            Ready to take the next step?
          </h3>
          <span
            class="sc-dAlyuH ixzeCR"
          >
            Create an account or sign in.
          </span>
          <p
            class="sc-hmdomO hwARNG"
          >
            By creating an account or logging in, you understand and agree to our Terms. You also acknowledge our Cookie and Privacy policies. You will receive marketing messages from Indeed and may opt out at any time by following the unsubscribe link in our messages, or as detailed in our terms.
          </p>
          <div
            class="sc-aXZVg cSsmnn"
          >
            <p
              class="sc-hmdomO hdkNeV"
            >
              <a
                class="sc-kOHTFB eYqJno"
                href="#"
              >
                Terms
              </a>
            </p>
            <p
              class="sc-hmdomO hdkNeV"
            >
              <a
                class="sc-kOHTFB eYqJno"
                href="#"
              >
                Cookie
              </a>
            </p>
            <p
              class="sc-hmdomO hdkNeV"
            >
              <a
                class="sc-kOHTFB eYqJno"
                href="#"
              >
                Privacy
              </a>
            </p>
          </div>
          <hr
            class="sc-fUnMCh jDzUIp"
          />
          <div
            class="sc-aXZVg fMjfal"
          >
            <div>
              <div
                class="sc-aXZVg kiRGJS"
              >
                <label
                  for="username"
                >
                  <span
                    class="sc-aXZVg kiRGJS sc-koXPp iPqYIw sc-koXPp iPqYIw"
                  >
                    <span
                      class="sc-dAlyuH fYJTzU"
                    >
                      Username
                    </span>
                  </span>
                </label>
              </div>
            </div>
            <div
              style="position:relative"
            >
              <input
                aria-invalid="false"
                class="sc-bmzYkS cwqa-DS"
                id="username"
                name="username"
                type="text"
                value=""
              />
              <div
                class="sc-iHGNWf jujJoU"
              />
            </div>
          </div>
          <div
            class="sc-aXZVg eyAcEv"
          >
            <div>
              <a
                class="sc-fPXMVe feVztn"
                direction="right"
                href="#"
              >
                <span
                  class="sc-aXZVg kiRGJS"
                >
                  <span>
                    Username forgot ?
                  </span>
                  <span
                    class="sc-gFqAkR eQHOvT"
                    direction="right"
                  >
                    <svg
                      class="sc-imWYAI sc-jXbUNg jJCqOQ lmNWrC"
                      size="16"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M9.18323023,17.7940812 C8.8825106,17.4424627 9.04504427,17.0896181 9.24527574,16.8868854 L13.740553,11.9968161 L9.24527573,7.11722187 C9.08145209,6.93839231 8.81996721,6.49053328 9.18323023,6.16524043 C9.54649326,5.83994757 9.91610528,6.06520964 10.0799289,6.24327169 L15.8766027,11.6348225 C16.0411324,11.8136521 16.0411324,12.1060729 15.8766027,12.2849025 C15.8758965,12.2849025 10.079929,17.7940809 10.079929,17.7940809 C9.93504267,17.9792355 9.48394985,18.1456996 9.18323023,17.7940812 Z"
                      />
                    </svg>
                  </span>
                </span>
              </a>
            </div>
            <div>
              <a
                class="sc-fPXMVe feVztn"
                direction="right"
                href="#"
              >
                <span
                  class="sc-aXZVg kiRGJS"
                >
                  <span>
                    Having issues submit a ticket
                  </span>
                  <span
                    class="sc-gFqAkR eQHOvT"
                    direction="right"
                  >
                    <svg
                      class="sc-imWYAI sc-jXbUNg jJCqOQ lmNWrC"
                      size="16"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M9.18323023,17.7940812 C8.8825106,17.4424627 9.04504427,17.0896181 9.24527574,16.8868854 L13.740553,11.9968161 L9.24527573,7.11722187 C9.08145209,6.93839231 8.81996721,6.49053328 9.18323023,6.16524043 C9.54649326,5.83994757 9.91610528,6.06520964 10.0799289,6.24327169 L15.8766027,11.6348225 C16.0411324,11.8136521 16.0411324,12.1060729 15.8766027,12.2849025 C15.8758965,12.2849025 10.079929,17.7940809 10.079929,17.7940809 C9.93504267,17.9792355 9.48394985,18.1456996 9.18323023,17.7940812 Z"
                      />
                    </svg>
                  </span>
                </span>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
